doc/devel/gitwash/development_workflow.rst

```diff
@@ -172,6 +172,71 @@ When you are ready to ask for someone to review your code and consider a merge:
    pull request message.  This is still a good way of getting some preliminary
    code review.
 
+Update your pull request
+========================
+
+Often, you'll want to update your contribution based on the feedback in the
+pull request. While it's possible to simply push additional commits to your
+feature branch, this will clutter the history. Not every step in the
+development of a feature is worth to be recorded. In contrast, a clean
+history with one or a few thematically separted commits helps to keep it
+more understandable. Most small features will do with a single commit.
+
+There are two techniques to achieve this.
+
+Ammend an existing commit
+   Use this if you have don't need to keep track of the incremental changes
+   to your feature.
+Squash multiple commits
+   Use this to cleanup the history when you've already made multiple commits
+   to your feature branch.
+
+Both techniques rewrite the history of your feature branch. While generally
+one has to be careful with rewriting the history, this is intended and
+perfectly fine for updates to your feature branch.
+
+The following descriptions apply the techniques to the last commits. This is
+slightly simpler than the general case, but the most common from when
+updating pull requests.
+
+Ammending the last commit
+-------------------------
+You have pushed a single commit for your feature and now want to update it.
+
+On your feature branch::
+
+   # make your updates
+   git commit --all --amend --no-edit
+   git push --force
+
+`--amend` means that the last commit will be rewritten to include the changes
+instead of creating a new commit. If `--no-edit` is used the existing commit
+message will be used. Otherwise, you'll be asked if you want to modify the
+commit message.
+
+Since you are overwriting the last commit with the rewritten version, you
+have to add `--force` to the push command. Github will automatically update
+your pull request after the push.
+
+Squashing the last N commits
+----------------------------
+You have N commits in your feature branch (e.g. N=3) and want to turn them
+into a single commit.
+
+On your feature branch::
+
+    git reset --soft HEAD~3
+    git commit
+    git push --force
```

Comment:
does this achieve the same thing as: 

```
git checkout feature-branch
git rebase -i HEAD~8 # opens text editor, check your gitconfig
# use the fixup option on all but the first line
# save and exit text editor (for vim that's ":wq"
git push origin feature-branch --force
```

Reply:
Yeah it does.  But I think it requires one fewer editor opens.    I somewhat prefer the rebase option.

Reply:
I think so, but you're never quite sure with git. I took this from https://stackoverflow.com/a/5201642/2726279

Reply:
In my experience rebasing is more difficult to understand than soft reset. That's why I chose this approach. I still feel like I cannot explain rebasing reasonably simple.